backend/services/userProfile.service.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable func-names */
+const mongoose = require('mongoose');
+const UserProfile = require('../models/userProfile.model');
+const modificationLogService = require('./modificationLog.service');
+
+const UserProfileService = {};
+
+UserProfileService.createUser = async function (userProfileData, authorEmail) {
+    
+    let newRecord = null;
+
+    const inputData = userProfileData;
+
+    if(Object.prototype.hasOwnProperty.call(inputData, "signupEmail") &&
+        inputData.signupEmail) {
+        inputData.signupEmail = userProfileData.signupEmail.toLowerCase();
+    }
+
+    await mongoose.connection.transaction(async () => {
```

Comment:
Please, wrap all mongoose calls in `try/catch` blocks. In `catch` block throw `DatabaseError` from #430 setting error message from mongoose error.
You can implement a solution for #430 in this PR.

Reply:
I've wrapped the calls in try/catch and throw a DatabaseError in the catch. Because mongoose already throws a validation error when required signupEmail is not present, this gets wrapped in our custom DatabaseError class. Let me know if you want something else to happen.

Reply:
Unless you want to differentiate that error in the catch, and throw a custom ValidationError code....
8:01
we could do
catch(e if e instanceof ReferenceError) {
    // Handle reference errors here
}
8:01
and rethrow a custom ValidationError

Reply:
We need to use a lot of custom validation in our application: #431 
So to not create a mess in controllers that should send responses depending on services' success or error it's better to catch all errors from other libraries and re-throw custom ones only.